### PR TITLE
Complex message added for add to cart success message

### LINKED ISSUE #29097: [Issue] Complex message added for add to cart success message
This issue is automatically created based on existing pull request: magento/magento2#27977: Complex message added for add to cart success message

---------

**Preconditions:**

1. Have at least 1 simple product;
2. Create a new customer;
3. One created order;

**Steps to reproduce:**

1. Add the product to cart from the customer account sidebar (**_Recently ordered product)_**

**Actual Resut:** :heavy_multiplication_x: The add to cart success message came up without the shopping cart link attached to it.

**Expected Result:** :heavy_check_mark: The add to cart success message should come up with the shopping cart link attached to it.

2. Add the product to cart from the customer account sidebar (**_Wishlist)_**

**Actual Result:** :heavy_multiplication_x: The add to cart success message came up without the shopping cart link attached to it.

**Expected Result:** :heavy_check_mark: The add to cart success message should come up with the shopping cart link attached to it.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by ajithkumar-maragathavel**
### Description (*)

Usually when we add a product from the product listing page or product description page.
The add to cart success message will come up with the shopping cart link attached to it.

Please refer the below screenshot:
success-message-with-link

But when we add the products from the customer account sidebar (_Recently ordered product or Wishlist_), The add to cart success message will not come up with the shopping cart link.

Please refer the below screenshot:
success-message-without-link

Please refer to the customer account sidebar screenshot below:
sidebar-add-to-cart

_**Therefore I have replaced the existing message with the add to cart success complex message with the shopping cart link.**_

### Manual testing scenarios (*)

1. Added a product from the customer account sidebar to the cart and verified the success message comes up with the shopping cart link.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29097: Complex message added for add to cart success message

### CODE REVIEW

**Comment by srsathish92** [comment]
Hello @ajithkumar-maragathavel , thank you for the contribution.

Please note, due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) the changes must be covered by tests. Could you please cover your fix by the appropriate unit test?

Thank you!

**Comment by VladimirZaets** [comment]
@magento create issue

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @ajithkumar-maragathavel,
Could you resolve merge conflict & cover your change with some kind of test?

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

File: dev/tests/integration/testsuite/Magento/Wishlist/Controller/Index/CartTest.php [hunk 1]
```diff
@@ -69,7 +69,8 @@ public function testAddSimpleProductToCart(): void
         $item = $this->getWishlistByCustomerId->getItemBySku(1, 'simple-1');
         $this->assertNotNull($item);
         $this->performAddToCartRequest(['item' => $item->getId(), 'qty' => 3]);
-        $message = sprintf('You added %s to your shopping cart.', $item->getName());
+        $message = sprintf("\n" . 'You added %s to your ' .
```
Comment (ihor-sviziev) [inline dev/tests/integration/testsuite/Magento/Wishlist/Controller/Index/CartTest.php:72]: Why do we need new like character here at the beginning ?

Comment (engcom-Echo) [reply dev/tests/integration/testsuite/Magento/Wishlist/Controller/Index/CartTest.php:72]: I think this is due to the using of "addComplexSuccessMessage".
It looks like we can see in 
app/code/Magento/Checkout/Controller/Cart/Add.php
image
dev/tests/integration/testsuite/Magento/Checkout/Controller/CartTest.php
image

Additional research is needed here.

Comment (ihor-sviziev) [reply dev/tests/integration/testsuite/Magento/Wishlist/Controller/Index/CartTest.php:72]: Ah, inside test it's ok. No need to investigate it

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7949](https://jira.corp.magento.com/browse/ENGCOM-7949) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Preconditions:**

1. Have at least 1 simple product;
2. Create a new customer;
3. One created order;

**Manual testing scenario:**

1. Add the product to cart from the customer account sidebar (**_Recently ordered product)_**

**Before:** :heavy_multiplication_x: The add to cart success message came up without the shopping cart link attached to it.



**After:** :heavy_check_mark: The add to cart success message came up with the shopping cart link attached to it.



2. Add the product to cart from the customer account sidebar (**_Wishlist)_**

**Before:** :heavy_multiplication_x: The add to cart success message came up without the shopping cart link attached to it.



**After:** :heavy_check_mark: The add to cart success message came up with the shopping cart link attached to it.



Also was tested on the Blank theme and everything works as expected.

**Pull request merged**